File: Gdc.Scd.BusinessLogicLayer/Interfaces/ITableViewService.cs
using System.Collections.Generic;
using System.IO;
using System.Threading.Tasks;
using Gdc.Scd.BusinessLogicLayer.Entities;
using Gdc.Scd.Core.Dto;
using Gdc.Scd.Core.Entities;
using Gdc.Scd.Core.Entities.QualityGate;
using Gdc.Scd.Core.Entities.TableView;

namespace Gdc.Scd.BusinessLogicLayer.Interfaces
{
    public interface ITableViewService
    {
        Task<IEnumerable<Record>> GetRecords();

        Task<QualityGateResultSet> UpdateRecords(IEnumerable<Record> records, ApprovalOption approvalOption);

        Task<TableViewInfo> GetTableViewInfo();

        Task<DataInfo<HistoryItemDto>> GetHistory(CostElementIdentifier costElementId, IDictionary<string, long> coordinates, QueryInfo queryInfo = null);

        Task<Stream> ExportToExcel();
    }
}

File: Gdc.Scd.Web.Client/src/TableView2/Services/ITableViewService.ts
import { TableViewInfo, QualityGateResultSet } from "../../TableView/States/TableViewState";
import { CostMetaData } from "../../Common/States/CostMetaStates";
import { TableViewRecord } from "../../TableView/States/TableViewRecord";
import { ApprovalOption } from "../../QualityGate/States/ApprovalOption";

export interface ITableViewService {
    getMeta(): Promise<CostMetaData>;

    getSchema(): Promise<TableViewInfo>;

    getUrl(): string;

    updateRecords(records: TableViewRecord[], approvalOption: ApprovalOption): Promise<QualityGateResultSet>;
}